ocs_ci/deployment/deployment.py

```diff
@@ -174,23 +196,296 @@ def deploy_cluster(self, log_cli_level="DEBUG"):
                 tmp_path, master_zones, worker_zones, x_addr_list, arbiter_zone
             )
 
+        # Multicluster operations
+        if config.multicluster:
+            # disable defaultsources: required for ACM based installs in DR usecase
+            # Need to run on all the clusters
+            for i in range(config.nclusters):
+                config.switch_ctx(i)
+                disable_default_sources()
+            config.reset_ctx()
+
+            # Configure submariner only on non-ACM clusters
+            submariner = Submariner()
+            submariner.deploy()
+
         if not config.ENV_DATA["skip_ocs_deployment"]:
-            try:
-                self.deploy_ocs()
-
-                if config.REPORTING["collect_logs_on_success_run"]:
-                    collect_ocs_logs("deployment", ocp=False, status_failure=False)
-            except Exception as e:
-                logger.error(e)
-                if config.REPORTING["gather_on_deploy_failure"]:
-                    # Let's do the collections separately to guard against one
-                    # of them failing
-                    collect_ocs_logs("deployment", ocs=False)
-                    collect_ocs_logs("deployment", ocp=False)
-                raise
+            for i in range(config.clusters):
+                if config.multicluster:
+                    if config.acm_index == i:
+                        continue
+                    config.switch_ctx(i)
+                try:
+                    self.deploy_ocs()
+
+                    if config.REPORTING["collect_logs_on_success_run"]:
+                        collect_ocs_logs("deployment", ocp=False, status_failure=False)
+                except Exception as e:
+                    logger.error(e)
+                    if config.REPORTING["gather_on_deploy_failure"]:
+                        # Let's do the collections separately to guard against one
+                        # of them failing
+                        collect_ocs_logs("deployment", ocs=False)
+                        collect_ocs_logs("deployment", ocp=False)
+                    raise
+            config.reset_ctx()
+
+            # Run ocs_install_verification here only in case of multiclsuter.
+            # For single cluster, test_deployment will take care.
+            if config.multicluster:
+                for i in range(config.multicluster):
+                    if config.acm_index == i:
+                        continue
+                    else:
+                        config.switch_ctx(i)
+                        ocs_registry_image = config.DEPLOYMENT["ocs_registry_image"]
+                        ocs_install_verification(ocs_registry_image=ocs_registry_image)
+                config.reset_ctx()
         else:
             logger.warning("OCS deployment will be skipped")
 
+        # Multicluster: Install ODF multicluster orchestrator only on hub
+        if config.multicluster:
```

Comment:
I think that this code for multicluster should be moved out of the deploy_cluster method and you should just call the new method you will move the code to from here.

Reply:
done, separated the class for multicluster dr related deployment code.